### PR TITLE
services: Move cfg to the top of the module map

### PR DESCRIPTION
**Pull request by GPUCode**
When using LLE service modules, AttemptLLE calls AppLoader_NCCH::Load which uses the cfg module to set the correct region. This means that using any LLE modules before cfg would assert as the module has not been initialized yet.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6943)

### CODE REVIEW

File: src/core/hle/service/service.cpp [hunk 1]
```diff
@@ -58,7 +58,8 @@
 namespace Service {
 
 const std::array<ServiceModuleInfo, 41> service_module_map{
-    {{"FS", 0x00040130'00001102, FS::InstallInterfaces},
+    {{"CFG", 0x00040130'00001702, CFG::InstallInterfaces},
```
Comment (Steveice10) [inline src/core/hle/service/service.cpp:61]: CFG should probably not be at the very top, it should at least be below the base kernel modules (fs, pm, ldr, pxi)

**Review by Steveice10 — COMMENTED** [review]
FYI it’s also possible to just access CFG data without the module loaded into system: https://github.com/citra-emu/citra/blob/master/src/citra_qt/configuration/configure_system.cpp#L303

**Comment by GPUCode** [comment]
> FYI it’s also possible to just access CFG data without the module loaded into system: https://github.com/citra-emu/citra/blob/master/src/citra_qt/configuration/configure_system.cpp#L303

From what I saw, using a dummy cfg is only for reading and not writing? Is it also applicable in this case?

**Comment by Steveice10** [comment]
> > FYI it’s also possible to just access CFG data without the module loaded into system: https://github.com/citra-emu/citra/blob/master/src/citra_qt/configuration/configure_system.cpp#L303
> 
> From what I saw, using a dummy cfg is only for reading and not writing? Is it also applicable in this case?



> > FYI it’s also possible to just access CFG data without the module loaded into system: https://github.com/citra-emu/citra/blob/master/src/citra_qt/configuration/configure_system.cpp#L303
> 
> From what I saw, using a dummy cfg is only for reading and not writing? Is it also applicable in this case?

I see what you mean, so I guess it’s not applicable here.

However I do still think the NCCH loader should not be dependent on the module load order in the first place. What if we’re trying to load an LLE CFG module, for example?

I would rather see us reverse the dependency in some way, where it’s the CFG module that fetches the auto region from elsewhere.

**Comment by GPUCode** [comment]
I also agree to this, the PR is just a hotfix to the real issue. I'll mark this as draft until I can came up with a better solution

**Comment by Steveice10** [comment]
> I also agree to this, the PR is just a hotfix to the real issue. I'll mark this as draft until I can came up with a better solution

Could probably have CFG use system.GetAppLoader() and expose the preferred region to it through the app loader interface.

**Comment by GPUCode** [comment]
I have pushed an attempt at proposed solution. It breaks home menu booting so needs more investigation

**Comment by GPUCode** [comment]
Also passing system as parameter to Module causes boost compile errors, would need to remove global state before merging this

**Comment by Steveice10** [comment]
> Also passing system as parameter to Module causes boost compile errors, would need to remove global state before merging this

Plenty of other modules have the system, I think you just need to adjust the boost serialization macros

**Comment by Adamillo** [comment]
Any updates on this PR?

**Pull request closed without merging**